pkg/apis/pipeline/v1alpha1/pipeline_types.go

```diff
@@ -167,6 +167,16 @@ func (pt PipelineTask) Deps() []string {
 		for _, rd := range cond.Resources {
 			deps = append(deps, rd.From...)
 		}
+		for _, param := range cond.Params {
+			expressions, ok := v1beta1.GetVarSubstitutionExpressionsForParam(param)
+			if ok {
+				if resultRefs, err := v1beta1.NewResultRefs(expressions); err == nil {
+					for _, resultRef := range resultRefs {
+						deps = append(deps, resultRef.PipelineTask)
+					}
+				}
+			}
+		}
```

Comment:
are there unit tests for this file? this seems like it would be nice to cover with unit tests

Reply:
looks like the unit test coverage went down significantly:

![image](https://user-images.githubusercontent.com/432502/78833887-a60df800-79bb-11ea-983e-156b94e7c947.png)

Reply:
Added dag tests for conditions dependencies. I believe v1beta1 pipeline types might need the same changes for dependencies.